### PR TITLE
Added instructions for final ChatGPT section.

### PR DESCRIPTION
**Pull request by mwhittaker**
REVIEWERS: visit https://github.com/ServiceWeaver/workshops/tree/chatgpt for a rendered version of this PR.

### CODE REVIEW

File: 10/README.md [hunk 1]
```diff
@@ -1,5 +1,184 @@
...
+
+Next, visit https://platform.openai.com/account/api-keys and click "Create new
+secret key". When prompted, enter a name for the secret key (e.g., "emojis") and
+click "Create secret key". Finally, OpenAI will generate and show you your
+secret key. Copy this key and store it somewhere, as you won't be able to view
+the secret key again once you click "Done". If you forget to copy the key,
+simply delete it and create a new one.
+
+⚠️ Note that this key is secret. Don't share it with anyone and **be careful not
+push it to a public repository**. ⚠️
```
Comment (rgrandl) [inline 10/README.md:29]: nit: not to push it

File: 10/README.md [hunk 2]
```diff
@@ -1,5 +1,184 @@
...
+type ChatGPT interface {
+    // Complete returns the ChatGPT completion of the provided prompt.
+    Complete(ctx context.Context, prompt string) (string, error)
+}
+```
+
+The `Complete` method receives a prompt (e.g., `"1 + 1 = "`) and returns
+ChatGPT's completion of the prompt (e.g., `"2"`). To implement the method, we
+recommend using [OpenAI's officially recommended Go library][openai_go]
+https://github.com/sashabaranov/go-openai. In the implementation of the
```
Comment (rgrandl) [inline 10/README.md:47]: remove?

**Review by rgrandl — APPROVED** [review]
Looks amazing! Can't wait to try it.

File: 10/README.md [hunk 3]
```diff
@@ -1,5 +1,184 @@
 # ChatGPT
 
-TODO(mwhittaker): Write this part.
+In this part, you will improve your emoji search engine using [OpenAI's GPT
+API][gpt_api]. Thus far, your emoji search engine implements *lexical search*.
+Given a query, it looks for all emojis with labels that match the literal
+strings in the query. In this part, you will use ChatGPT to implement *semantic
+search*. The search engine will return emojis that match the meaning of the
+query.  Specifically, when you get a query, you will ask ChatGPT for emojis
+related to the query. For the query "summer vibes", for example, ChatGPT returns
```
Comment (spetrovic77) [inline 10/README.md:9]: Nice example :)

File: 10/README.md [hunk 4]
```diff
@@ -1,5 +1,184 @@
...
+emojis like 🌞, 🌴, and 🍹. These emojis match the meaning of "summer vibes"
+even though they aren't labeled "summer" or "vibes".
+
+## OpenAI Account
+
+First, visit https://platform.openai.com/signup and create an OpenAI account.
+OpenAI's GPT API is not free to use, but as of this writing, new OpenAI accounts
+automatically receive [$5 in free credit][pricing]. If you're reading this and
+the promotion no longer exists, you'll have to set up [billing][] and purchase
```
Comment (spetrovic77) [inline 10/README.md:18]: Missing link?

File: 10/README.md [hunk 5]
```diff
@@ -1,5 +1,184 @@
...
+```
+
+You can curl the `/search_chatgpt` endpoint directly:
+
+```console
+$ curl localhost:9000/search_chatgpt?q=happy
+["😀","😁","😃","😄","😊","😍","😎","🤗","😻","🌞","🎉","🎊","🎁","🎈","💐","👍","✨","🌟","💫","🌈"]
+```
+
+Though, we strongly recommend you use the web UI provided in [Part 4](../04).
```
Comment (spetrovic77) [inline 10/README.md:169]: Just checking if this UI was introduced in Part 5?

**Review by spetrovic77 — APPROVED** [review]
This is awesome. I think we need a separate workshop just for this. 😄 But this is a great exercise for folks to try back home.

File: 10/README.md (hunk 3 above)
Comment (mwhittaker) [reply 10/README.md:9]: Thanks 🌞 🌴 🍹

File: 10/README.md (hunk 4 above)
Comment (mwhittaker) [reply 10/README.md:18]: Done.

File: 10/README.md (hunk 1 above)
Comment (mwhittaker) [reply 10/README.md:29]: D'oh. Fixed.

File: 10/README.md (hunk 2 above)
Comment (mwhittaker) [reply 10/README.md:47]: Reworded.

File: 10/README.md (hunk 5 above)
Comment (mwhittaker) [reply 10/README.md:169]: Should be Part 4.

**Review by mwhittaker — COMMENTED** [review]
Thanks Robert and Srdjan!

**Pull request merged**